connector/connectortest/router.go

```diff
@@ -0,0 +1,108 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package connectortest // import "go.opentelemetry.io/collector/connector/connectortest"
+
+import (
+	"errors"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/connector"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/consumer/consumertest"
+	"go.opentelemetry.io/collector/internal/fanoutconsumer"
+)
+
+var errTooFewConsumers = errors.New("router requires a minimum of 2 consumers")
+
+type TracesRouterTestOption struct {
+	id   component.ID
+	cons consumer.Traces
+}
+
+// WithNopTracesSink creates a nop consumer for a connector.TracesRouter
+func WithNopTracesSink(id component.ID) TracesRouterTestOption {
+	return TracesRouterTestOption{id: id, cons: consumertest.NewNop()}
+}
+
+// WithTracesSink adds a consumer to a connector.TracesRouter
+func WithTracesSink(id component.ID, sink *consumertest.TracesSink) TracesRouterTestOption {
```

Comment:
Why not just `WithTraces` and accept a `consumer.Traces`

Reply:
This was my suggestion. The idea was to limit the usefulness of the TracesRouter to tests only. 

If we accept any `consumer.Traces`, then we might want to just export fanoutconsumer and let people build routers directly.

Reply:
May not be a bad idea (not the whole `fanoutconsumer`) only the router part correct? You don't need the `fanoutconsumer.NewLogs`.

Reply:
Is there any reason why someone would need to create a router other than for testing purposes?

Reply:
@bogdandrutu, WDYT?